proxy/http/HttpSM.cc

```diff
@@ -7618,6 +7616,8 @@ HttpSM::do_redirect()
     return;
   }
 
+  redirection_tries++;
+
```

Comment:
Right below here, we repeat the test on ``number_of_redirections`` with ``is_redirect_required(). The condition is different however, since this code will not redirect with ``number_of_redirections`` and ``is_redirect_required()`` will.

I don't think that this is the right place to increment ``redirection_tries``, though it is an improvement. How about incrementing it in ``redirect_request()``? Or at least defer until we know we are actually going to call ``redirect_request``.
 
We should restructure this function like this:
```C
if (!is_redirect_required()) {
    tunnel.deallocate_redirect_post_buffers();
    enable_redirection = false;
    return;
}

...
```